components/ui-api-layer/internal/domain/remoteenvironment/re_service.go

```diff
@@ -67,6 +69,47 @@ func newRemoteEnvironmentService(client remoteenvironmentv1alpha1.Applicationcon
 	}, nil
 }
 
+func (svc *remoteEnvironmentService) Create(name string, description string, labels map[string]interface{}) (*v1alpha1.RemoteEnvironment, error) {
+	return svc.client.RemoteEnvironments().Create(&v1alpha1.RemoteEnvironment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RemoteEnvironment",
+			APIVersion: "applicationconnector.kyma-project.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.RemoteEnvironmentSpec{
+			Labels:      svc.convertJsonToLabels(labels),
+			Description: description,
+			Services:    []v1alpha1.Service{},
+		},
+	})
+}
+
+func (svc *remoteEnvironmentService) Update(name string, description string, labels map[string]interface{}) (*v1alpha1.RemoteEnvironment, error) {
+	for i := 0; i < maxUpdateRetries; i++ {
+		re, err := svc.Find(name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "while getting RemoteEnvironment [%s]", name)
+		}
+		re.Spec.Description = description
+		re.Spec.Labels = svc.convertJsonToLabels(labels)
+
```

Comment:
IMO with switch statement, it's more readable
```go
		updated, err := svc.client.RemoteEnvironments().Update(re)
		switch {
		case err == nil:
			return updated, nil
		case apiErrors.IsConflict(err):
			continue
		default:
			return nil, errors.Wrapf(err, "while updating RemoteEnvironment [%s]", name)
		}
```

or to be a Go idiomatic, then this function should be refactored and have the happy path always on the left side.